publication/src/main/scala/hmda/publication/reports/aggregate/AggregateReport.scala

```diff
@@ -13,3 +19,12 @@ trait AggregateReport {
 
   val reportType: ReportTypeEnum = Aggregate
 }
+
+abstract class A5XReportCreator {
+  def filters(lar: LoanApplicationRegisterQuery): Boolean
+
+  def generate[ec: EC, mat: MAT, as: AS](
```

Comment:
This method naming seems a little confusing to me (in `A52` and `A53`, we call `generate` in another `generate` method).  Could we make these method names more specific?